igcollect/switch.py

```diff
@@ -281,42 +292,15 @@ def get_laggs(snmp, model):
     return None
 
 
-def standarize_portname(port_name):
+def standarize_portname(port_name, model):
     """ Return a Graphite-compatible port name or None if name
         can't be translated
     """
-
-    if re.match('\A(Po|Trk)[0-9]+\Z', port_name):
-        # Dell or HP LAGG
-        return port_name
-    if re.match('\A[0-9]+\Z', port_name):
-        # HP normal port
-        return port_name
-    if re.match('\A[0-9]:[0-9]+\Z', port_name):
-        # Extreme normal port
-        return port_name.replace(':', '_')
-    if re.match('\A(Gi|Te)[0-9]/[0-9]/[0-9]+\Z', port_name):
-        # Dell normal port
-        return port_name.replace('/', '_')
-    if re.match('\A(Fa|Gi|Tu)[0-9]/[0-9]+\Z', port_name):
-        # Cisco ports
-        return port_name.replace('/', '_')
-    if re.match('\A(Fa|Gi|Tu)[0-9]+\Z', port_name):
-        # Cisco tunnels
-        return port_name
-    if re.match('\A(swp)[0-9]+\Z', port_name):
-        # Cumulus ports
-        return port_name
-    g = re.match(
-        '\A(TenGigabitEthernet|fortyGigE) ([0-9]+/[0-9]+)\Z',
-        port_name
-    )
-    if g:
-        # Force 10 MXL port
-        return g.group(2).replace('/', '_')
-    if re.match('\Aethernet[0-9]+/[0-9]+\Z', port_name):
-        # Netiron ports
-        return port_name.replace('/', '_')
+    r = PORT_REGEXP[model].match(port_name)
+    if r:
+        g =  r.groupdict()['port']
+        if g:
```

Comment:
This if doesn't make sense. 
If the group is not matched and you try to fetch it, you get an IndexError exception.

Reply:
It does again after changing `r.groupdict()['port']` to `r.group('port')`

Reply:
Right, I misunderstood the docs. After changing this to r.group(), it returns None instead of the Exception.

Reply:
Fixed.